internal/client/cloud_client.go

```diff
@@ -91,8 +91,52 @@ func (c *CloudClient) Do(req *http.Request) (*ResponseWithETag, error) {
 		return nil, err
 	}
 
-	// Extract the ETag if it exists
-	etag := resp.Header.Get("ETag")
+	// Extract the ETag - check multiple possible header names
+	var etag string
+
+	// Check standard ETag header (case-insensitive)
+	etag = resp.Header.Get("ETag")
+
+	// If not found, check for other variations
+	if etag == "" {
+		etag = resp.Header.Get("etag")
+	}
+	if etag == "" {
+		etag = resp.Header.Get("Etag")
+	}
+
+	// Sometimes ETags are returned with quotes, clean those if present
+	if len(etag) > 0 && (etag[0] == '"' && etag[len(etag)-1] == '"') {
+		etag = etag[1 : len(etag)-1]
+	}
```

Comment:
This is also surprising to me - when/how was it happening? It might be something to fix in the upstream rather than here.